core/src/main/scala/sttp/openai/requests/completions/chat/message/Tool.scala

```diff
@@ -16,6 +20,31 @@ object Tool {
     */
   case class FunctionTool(description: String, name: String, parameters: Map[String, Value]) extends Tool
 
+  case class SchematizedFunctionTool(description: String, name: String, parameters: Schema) extends Tool
```

Comment:
some scalados here would be nice, specifying when to use `FuntionTool`, and when this class, and in which API